### PR TITLE
Sun Vector Tests and code cleanup

### PR DESCRIPTION
**Pull request by thetazero**
Results line up closely when compared to satellite dynamics

### CODE REVIEW

File: state_machine/unit_tests/test_igrf.py [hunk 1]
```diff
@@ -31,28 +28,28 @@ def test_matches_IGRF13(self):
         # using pyIGRF.py
```
Comment (jbwillis) [inline state_machine/unit_tests/test_igrf.py:28]: Can't comment there for some reason but line 21 is another duplicate use of EARTH_RADIUS. This should be referenced from one single place in all of the code.

**Review by jbwillis — CHANGES_REQUESTED** [review]

File: state_machine/unit_tests/test_sun_position.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+                        -5.302933759398837e7]
+            },
+        ]
+        sum_angle = 0.0
+        sum_dif = 0.0
+        for test in tests:
+            angle, dif = similar(approx_sun_position_ECI(test["time"]), test["ref"])
+            sum_angle += angle
+            sum_dif += dif
+        print("Average angle:", sum_angle / len(tests))
```
[the remark this answers is not in the document]
Comment (thetazero) [reply state_machine/unit_tests/test_sun_position.py]: Good catch

File: state_machine/unit_tests/test_igrf.py [hunk 3]
```diff
@@ -1,15 +1,12 @@
...
+from testutils import timestamp
 
 sys.path.insert(0, './state_machine/applications/flight')
 
 from lib.IGRF import igrf, igrf_eci
 
-def timestamp(year, month, day, hour=0, minute=0, second=0):
-    return dt(year, month, day, hour, minute, second, tzinfo=timezone.utc).timestamp()
-
 def assert_vector_almost_equal(a, b, assertLessEqual, angle_tolerance=5, nt_tolerance=3000):
```
[the remark this answers is not in the document]
Comment (thetazero) [reply state_machine/unit_tests/test_igrf.py]: Yup, good catch

File: state_machine/unit_tests/test_igrf.py (hunk 1 above)
Comment (thetazero) [reply state_machine/unit_tests/test_igrf.py:17]: I rather not do that for test cases, I want the inputs to the functions to be fixed regardless of if we update earth's radius. 
As for flight software I agree.

Comment (jbwillis) [reply state_machine/unit_tests/test_igrf.py:17]: That's fair. The way it is right now is not very clean though (especially after reviewing some of the other code and seeing many different EARTH_RADIUS definitions). This needs to be cleaned up somehow. If you want different instances of it, perhaps you could use EARTH_RADIUS_TEST_ONLY

File: state_machine/unit_tests/test_sun_position.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
+import unittest
+import sys
+from testutils import assert_vector_similar, timestamp
+
+sys.path.insert(0, './state_machine/applications/flight')
+
+from lib.sun_position import approx_sun_position_ECI
+
+class SunPosition(unittest.TestCase):
```
Comment (jbwillis) [inline state_machine/unit_tests/test_sun_position.py:9]: My general style (and I think this is more widely accepted) is to prepend `Test` to the names of classes that inherit from `unittest.TestCase`. So this should be `TestSunPosition`.

File: state_machine/unit_tests/test_sun_position.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        sum_dif = 0.0
+        for test in tests:
+            angle, dif = similar(approx_sun_position_ECI(test["time"]), test["ref"])
+            sum_angle += angle
+            sum_dif += dif
+        print("Average angle error:", sum_angle / len(tests))
+        print(f"Average absolute error {sum_dif / len(tests)/1e6} million km")
+
+
+x = SunPosition()
```
Comment (jbwillis) [inline state_machine/unit_tests/test_sun_position.py:87]: This and the following line are a bit weird to me. I believe they shouldn't be necessary if you run this with `python -m unittest`

**Review by jbwillis — CHANGES_REQUESTED** [review]

File: state_machine/unit_tests/test_sun_position.py (hunk 5 above)
Comment (thetazero) [reply state_machine/unit_tests/test_sun_position.py:87]: Yeah, I forgot to delete that.

**Pull request merged**